homeassistant/components/mqtt/scene.py

```diff
@@ -0,0 +1,145 @@
+"""Support for MQTT scenes."""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.components import mqtt, scene
+from homeassistant.components.scene import Scene
+from homeassistant.const import CONF_ICON, CONF_NAME, CONF_PAYLOAD_ON, CONF_UNIQUE_ID
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.reload import async_setup_reload_service
+from homeassistant.helpers.typing import ConfigType, HomeAssistantType
+
+from . import (
+    ATTR_DISCOVERY_HASH,
+    CONF_COMMAND_TOPIC,
+    CONF_QOS,
+    CONF_RETAIN,
+    DOMAIN,
+    PLATFORMS,
+    MqttAvailability,
+    MqttDiscoveryUpdate,
+)
+from .discovery import MQTT_DISCOVERY_NEW, clear_discovery_hash
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_NAME = "MQTT Scene"
+DEFAULT_RETAIN = False
+
+PLATFORM_SCHEMA = mqtt.MQTT_BASE_PLATFORM_SCHEMA.extend(
+    {
+        vol.Required(CONF_COMMAND_TOPIC): mqtt.valid_publish_topic,
+        vol.Optional(CONF_ICON): cv.icon,
+        vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+        vol.Optional(CONF_PAYLOAD_ON): cv.string,
+        vol.Optional(CONF_UNIQUE_ID): cv.string,
+        vol.Optional(CONF_RETAIN, default=DEFAULT_RETAIN): cv.boolean,
+    }
+).extend(mqtt.MQTT_AVAILABILITY_SCHEMA.schema)
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType, config: ConfigType, async_add_entities, discovery_info=None
+):
+    """Set up MQTT scene through configuration.yaml."""
+    await async_setup_reload_service(hass, DOMAIN, PLATFORMS)
+    await _async_setup_entity(hass, config, async_add_entities, discovery_info)
```

Comment:
The fourth parameter should be the config entry. Where is the platform loaded with the config entry as discovery_info?

Reply:
The same incorrect pattern is also present in `mqtt/switch.py` and `mqtt/vacuum.py`.

Reply:
Ok to fix later.

Reply:
Fixed in https://github.com/home-assistant/core/pull/44110